File: Gnome_sort/Go/jcla1/gnome_sort.go
package gnomeSort

// Gnome sort runs in O(n^2), but if the array
// is nearly sorted already it runs in about O(n)
// http://en.wikipedia.org/wiki/Gnome_sort
func GnomeSort(arr []int) []int {
    for pos := 1; pos < len(arr); pos++ {
        if arr[pos] >= arr[pos-1] {
            continue
        } else {
            arr[pos], arr[pos-1] = arr[pos-1], arr[pos]
            if pos > 1 {
                pos -= 1
            }
        }
        pos -= 1
    }

    return arr
}

File: Gnome_sort/Java/ghostsnstuff/GnomeSort.java
/*
 *  gnome sort
 *  ----------
 *  author: ghostsnstuff
 *  date: March 27, 2014
 *  src: http://en.wikipedia.org/wiki/Gnome_sort
 *  complexity:
 *   - worst case O(n^2)
 *   - best case O(n)
 *   - avg case O(n^2) 
 */
class GnomeSort {
  static int[] GnomeSort(int[] list) {
    int pos = 1;
    while(pos < list.length) {
      if(list[pos] >= list[pos-1]) {
        pos++;
      } else {
	int temp = list[pos];
	list[pos] = list[pos-1];
	list[pos-1] = temp;
	if(pos > 1) {
	  pos--;
	}
      }
    }
    return list;
  }		
  static void log(int[] list) {
    for(int i = 0; i < list.length; i++) {
      System.out.print(list[i]+" ");
    }
  }
}
